File: Source Code/backend/models/Fingerprint.js
const mongoose = require('mongoose');

const fingerprintSchema = new mongoose.Schema({
  fingerprint_id: {
    type: String,
    required: true,
    unique: true,
    index: true
  },
  user_id: {
    type: String,
    required: true,
    index: true
  },
  template_base64: {
    type: String,
    required: true
  },
  finger_position: {
    type: String,
    enum: ['thumb', 'index', 'middle', 'ring', 'pinky', 'unknown'],
    default: 'unknown'
  },
  hand: {
    type: String,
    enum: ['left', 'right', 'unknown'],
    default: 'unknown'
  },
  registered_at: {
    type: Date,
    default: Date.now
  }
});

fingerprintSchema.index({ user_id: 1, registered_at: -1 });

module.exports = mongoose.model('Fingerprint', fingerprintSchema);

File: Source Code/mobile-app/AuthenX/app/src/main/java/com/example/authenx/domain/model/Fingerprint.kt
package com.example.authenx.domain.model

import com.google.gson.annotations.SerializedName

data class Fingerprint(
    @SerializedName("_id")
    val id: String,
    @SerializedName("fingerprint_id")
    val fingerprintId: String,
    @SerializedName("user_id")
    val userId: String,
    @SerializedName("template_base64")
    val templateBase64: String,
    @SerializedName("finger_position")
    val fingerPosition: String, // "thumb", "index", "middle", "ring", "pinky", "unknown"
    val hand: String, // "left", "right", "unknown"
    @SerializedName("registered_at")
    val registeredAt: String
)

data class FingerprintsResponse(
    val success: Boolean,
    val message: String?,
    val data: List<Fingerprint>
)

data class AddFingerprintRequest(
    @SerializedName("user_id")
    val userId: String,
    @SerializedName("template_base64")
    val templateBase64: String,
    @SerializedName("finger_position")
    val fingerPosition: String = "unknown",
    val hand: String = "unknown"
)
